File: golang-code/ex12/produto.go
package main

import "fmt"

type Produto struct {
	Nome  string
	Preco float64
}

func (p Produto) Somar(outro Produto) Produto {
	return Produto{
		Nome:  p.Nome + " & " + outro.Nome,
		Preco: p.Preco + outro.Preco,
	}
}

func (p Produto) String() string {
	return fmt.Sprintf("%s - R$%.2f", p.Nome, p.Preco)
}

File: java-code/src/main/java/ex12/Produto.java
package ex12;

public class Produto {
    private String nome;
    private double preco;

    public Produto(String nome, double preco) {
        this.nome = nome;
        this.preco = preco;
    }

    public Produto somar(Produto outro) {
        return new Produto(this.nome + " e " + outro.nome, this.preco + outro.preco);
    }

    @Override
    public String toString() {
        return nome + " - R$" + String.format("%.2f", preco);
    }
}
